shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -313,6 +362,13 @@ impl CassandraSinkCluster {
             super::connection::receive(self.read_timeout, &self.failed_requests, responses_future)
                 .await?;
 
+        let _prepare_responses = super::connection::receive(
```

Comment:
We should return an error if cassandra returns something unexpected here.

Reply:
Yeah this part is a bit tricky, we return the response to PREPARE from the handshake connection. But if an error occurs on one of the nodes, should we use that response or the handshake one?

Reply:
![image](https://user-images.githubusercontent.com/5120858/192915457-9be989d9-6bfd-47df-8252-793e5e558c23.png)

Looking at it closer returning an error is too extreme.

When we send use statements to every open connection, we close any connections that return strange results as we can no longer uphold our invariants on those connections.
We should probably do that here as well.

In both of these cases we should also log the error, seems I forgot to do so with the use statements.

Reply:
We don't need to close the connection because the query will remain unprepared, so it will just attempt to reprepare it when the node is hit with an execute.

I'd be happy with either approach TBH, closing or not closing would both work.

Reply:
I believe this has been addressed, we are now logging errors for when the response to a prepare statement is not equal on every node.

Reply:
Yep looks good.
The logic is a bit wonky because it doesnt take into account the message sent to the control connection, but that will be cleared up once we change the prepare logic to not send to the control connection in a follow up PR